packages/flutter/test/widgets/sliver_constrained_cross_axis_test.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('SliverConstrainedCrossAxis basic test', (WidgetTester tester) async {
+    await tester.pumpWidget(_buildSliverConstrainedCrossAxis(extent: 50));
+
+    final RenderBox box = tester.renderObject(find.byType(Container));
+    expect(box.size.height, 100);
+    expect(box.size.width, 50);
+  });
+
+  testWidgets('SliverConstrainedCrossAxis updates correctly', (WidgetTester tester) async {
+    await tester.pumpWidget(_buildSliverConstrainedCrossAxis(extent: 50));
+
+    final RenderBox box1 = tester.renderObject(find.byType(Container));
+    expect(box1.size.height, 100);
```

Comment:
Can you also check the sliver geometry? Example: https://github.com/flutter/flutter/blob/e6257302dbccfd69cdb7719b856d4f95e840ad79/packages/flutter/test/widgets/sliver_fill_viewport_test.dart#L185

Reply:
Done!